frontend/src/scenes/dashboard/DashboardTemplateEditorLogic.ts

```diff
@@ -0,0 +1,157 @@
+            },
+        ],
+    }),
+    loaders(({ values }) => ({
+        dashboardTemplate: [
+            undefined as DashboardTemplateEditorType | undefined | null,
+            {
+                createDashboardTemplate: async (): Promise<DashboardTemplateEditorType | undefined> => {
+                    if (!values.dashboardTemplate) {
+                        lemonToast.error('Unable to create dashboard template')
+                        return
+                    }
+                    const response = await api.dashboardTemplates.create(values.dashboardTemplate)
+                    lemonToast.success('Dashboard template created')
+                    return response
+                },
+                getDashboardTemplate: async (id: string): Promise<DashboardTemplateType> => {
+                    const response = await api.dashboardTemplates.get(id)
+                    return response
+                },
+                updateDashboardTemplate: async (id: string): Promise<DashboardTemplateType | undefined> => {
+                    if (!values.dashboardTemplate) {
+                        lemonToast.error('Unable to update dashboard template')
+                        return
+                    }
+                    const response = await api.dashboardTemplates.update(id, values.dashboardTemplate)
+                    lemonToast.success('Dashboard template updated')
+                    return response
+                },
+                deleteDashboardTemplate: async (id: string): Promise<null> => {
+                    await api.dashboardTemplates.delete(id)
+                    lemonToast.success('Dashboard template deleted')
+                    return null // for some reason this errors when it's undefined instead
+                },
+            },
+        ],
+        templateSchema: [
+            undefined as Record<string, any> | undefined,
+            {
+                getTemplateSchema: async (): Promise<Record<string, any>> => {
+                    debugger
+                    const response = await api.dashboardTemplates.getSchema()
+                    return response
+                },
+            },
+        ],
+    })),
+    listeners(({ values, actions }) => ({
+        createDashboardTemplateSuccess: async () => {
+            actions.closeDashboardTemplateEditor()
+            dashboardTemplatesLogic.actions.getAllTemplates()
+        },
+        updateDashboardTemplateSuccess: async () => {
+            actions.closeDashboardTemplateEditor()
+            dashboardTemplatesLogic.actions.getAllTemplates()
+        },
+        deleteDashboardTemplateSuccess: async () => {
+            dashboardTemplatesLogic.actions.getAllTemplates()
+        },
+        closeDashboardTemplateEditor: () => {
+            actions.clear()
+        },
+        setDashboardTemplateId: async ({ id }) => {
+            if (id) {
+                await actions.getDashboardTemplate(id)
```

Comment:
This `await` doesn't really do anything. If you really want to wait for an action to finish all its connected listeners, there's `asyncActions`... but that's a bit of an anti pattern. Normally "dispatching an action" is a fast operation that just puts a message on the queue, so there's nothing you can wait for.